src/subscription.rs

```diff
@@ -95,14 +95,7 @@ impl<'a, T: Runtime> EventSubscription<'a, T> {
             if self.finished {
                 return None
             }
-            let change_set = match self.subscription.next().await {
-                Some(c) => c,
-                None => {
```

Comment:
The proper action should be to indicate that the subscription is dropped and need to restarted because after it returns `None` it shouldn't be polled anymore because the channel has been terminated.

If you poll if after `None` it will panic

Reply:
Which is exactly why I would expect it to return `None`, to indicate the stream has closed (much like with `StreamExt`).

Reply:
Alright, that makes sense to me but perhaps we should document it because it's not obvious to me ^^